packages/editor/src/components/provider/index.js

```diff
@@ -180,24 +184,32 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 					getRenderingMode,
 					__unstableIsEditorReady,
 				} = select( editorStore );
-				const { getEntitiesConfig } = select( coreStore );
+				const {
+					getEntitiesConfig,
+					getPostType,
+					hasFinishedResolution,
+				} = select( coreStore );
 
-				const postTypeObject = select( coreStore ).getPostType(
-					post.type
+				const postTypeSupports = getPostType( post.type )?.supports;
+				const hasLoadedPostObject = hasFinishedResolution(
+					'getPostType',
+					[ post.type ]
 				);
 
-				const _hasLoadedPostObject = select(
-					coreStore
-				).hasFinishedResolution( 'getPostType', [ post.type ] );
+				const _defaultMode = Array.isArray( postTypeSupports?.editor )
+					? postTypeSupports.editor.find(
+							( features ) => 'default_mode' in features
+					  )?.default_mode
+					: undefined;
+				const hasDefaultMode = RENDERING_MODES.includes( _defaultMode );
```

Comment:
Had to go with the client-side validation since it's not possible to validate arguments passed via [`add_post_type_support`](https://developer.wordpress.org/reference/functions/add_post_type_support/).